Singleton/src/Singleton.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Singleton.hpp"
+
+// Singleton factory
+Example* SingletonFactory::get() {
+    static Example e{};
+    return &e;
+};
+
+// Not a singleton factory
+Example* NotSingletonFactory::get() {
+    return new Example{};
+};
+
+// Singleton Tester
+template <typename T>
+bool SingletonTester::is_singleton(std::function<T* ()> factory)
```

Comment:
There are two options for fixing the linker errors:

- Move the definition in the header file
- Explicitly instantiate the template function in the .cpp file like so: `bool SingletonTester::is_singleton(std::function<MyClass* ()> factory)` where MyClass is the concrete type which can substitute T. If you pick option 2 and want to have multiple types which have to substitute T, then an explicit instantiation for each one is needed.

Reply:
Done, thank you.